File: CodeSignal/C++/Edge of the Ocean/adjacentElementsProduct.cpp
int adjacentElementsProduct(std::vector<int> inputArray) {
    int largestProduct = inputArray[0] * inputArray[1];
    for (int i = 0; i < inputArray.size() - 1; i++) {
        int newProduct = inputArray[i] * inputArray[i+1];
        if (largestProduct < newProduct) {
            largestProduct = newProduct;
        }
    }
    return largestProduct;
}

File: CodeSignal/Edge of the Ocean/py - 5:5/adjacentElementsProduct.py
def adjacentElementsProduct(inputArray):
    max_prod = inputArray[0] * inputArray[1]
    for i in range(len(inputArray)-1):
        prod = inputArray[i] * inputArray[i+1]
        if prod > max_prod:
            max_prod = prod
    return max_prod
